sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/UnusedPrivateMemberTest.cs

```diff
@@ -29,9 +29,640 @@ public class UnusedPrivateMemberTest
     {
         [TestMethod]
         [TestCategory("Rule")]
-        public void UnusedPrivateMember()
+        public void Field_Accessibility()
         {
-            Verifier.VerifyAnalyzer(@"TestCases\UnusedPrivateMember.cs", new UnusedPrivateMember());
+            Verifier.VerifyCSharpAnalyzer(@"
+public class PrivateMembers
+{
+    private int privateField; // Noncompliant {{Remove the unused private field 'privateField'.}}
+//  ^^^^^^^^^^^^^^^^^^^^^^^^^
+    private static int privateStaticField; // Noncompliant
+
+    private class InnerPrivateClass // Noncompliant
+    {
+        internal int internalField; // Noncompliant
+        protected int protectedField; // Noncompliant
+        protected internal int protectedInternalField; // Noncompliant
+        public int publicField; // Noncompliant
+        internal static int internalStaticField; // Noncompliant
+        protected static int protectedStaticField; // Noncompliant
+        protected internal static int protectedInternalStaticField; // Noncompliant
+        public static int publicStaticField; // Noncompliant
```

Comment:
If an inner class is unused, could we avoid reporting on any unused member in that type too reduce the noise?

Reply:
Sounds like a good candidate for a new ticket.